src/frontend/src/components/forms/fields/RelatedModelField.tsx

```diff
@@ -71,28 +79,49 @@ export function RelatedModelField({
   const [value, setValue] = useState<string>('');
   const [searchText, cancelSearchText] = useDebouncedValue(value, 250);
 
+  const [filters, setFilters] = useState<any>({});
+
+  const resetSearch = useCallback(() => {
+    setOffset(0);
+    setData([]);
+    dataRef.current = [];
+  }, []);
+
   // reset current data on search value change
   useEffect(() => {
-    dataRef.current = [];
-    setData([]);
-  }, [searchText]);
+    resetSearch();
+  }, [searchText, filters]);
 
   const selectQuery = useQuery({
-    enabled: !definition.disabled && !!definition.api_url && !definition.hidden,
+    enabled:
+      isOpen &&
+      !definition.disabled &&
+      !!definition.api_url &&
+      !definition.hidden,
     queryKey: [`related-field-${fieldName}`, fieldId, offset, searchText],
     queryFn: async () => {
       if (!definition.api_url) {
         return null;
       }
 
-      let filters = definition.filters ?? {};
+      let _filters = definition.filters ?? {};
 
       if (definition.adjustFilters) {
-        filters = definition.adjustFilters(filters);
+        _filters =
+          definition.adjustFilters({
+            filters: _filters,
+            data: form.getValues()
```

Comment:
Why do we need that data here? Can't we just hook the required values to a state and use them in the filter function?

Reply:
By passing the `data` object through, we ensure that we get a "fresh" set of data on every call, without needing to setup individual hooks from each other field we are interested in. Say that a related field wants to adjust the API filters based on the values of three other fields - that's three additional hooks, and a lot of additional boilerplate code.

This approach feels much cleaner and more consistent to me. There may be a performance tradeoff, but at a significant advantage in terms of reduced code complexity.

Reply:
Ok, makes sense.